core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -941,6 +941,14 @@ namespace kagome::consensus::grandpa {
              peer_id,
              msg);
 
+    // Skip message processing if same vote was already observed
+    if (votes_cache_.contains(msg)) {
+      return;
+    }
+    votes_cache_.put(msg);
+
+    if (not info.has_value() or not info->set_id.has_value()
```

Comment:
Copy of line 953. Need to remove it

Reply:
At least info is used before the definition

Reply:
No problem, in the line 952 it might be rewritten.

Reply:
It got here accidentally. Fixed now